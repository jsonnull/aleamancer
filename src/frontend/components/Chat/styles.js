// @flow
import styled from 'styled-components'

export const Container = styled.div`
  background-color: ${props => props.theme.background};
  width: 320px;
  overflow: hidden;
  border-radius: ${props => (props.isPinned ? '0' : '5px')};
  box-shadow: ${props =>
    props.isPinned ? '' : '0 3px 5px 0 rgba(0, 0, 0, 0.2)'};

  height: ${props => (props.isPinned ? '100%' : 'auto')};
  position: ${props => (props.isPinned ? 'relative' : 'absolute')};
  top: ${props => (props.isPinned ? '0' : 'auto')};
  right: ${props => (props.isPinned ? '0' : '1rem')};
  bottom: ${props => (props.isPinned ? '0' : '1rem')};

  display: flex;
  flex-direction: column;
`

export const Loading = styled.div`
  padding: 10px;
`

export const Error = Loading
