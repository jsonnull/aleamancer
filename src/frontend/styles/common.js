// @flow
export const fonts = {
  heading: 'Lora',
  body: `
    /* 1 */ -apple-system, BlinkMacSystemFont,
    /* 2 */ "Segoe UI", "Roboto", "Oxygen", "Ubuntu", "Cantarell", "Fira Sans", "Droid Sans",
    /* 3 */ "Helvetica Neue", sans-serif;
  `
}

export const fontSize = {
  small: '1.3rem',
  normal: '1.5rem',
  medium: '1.8rem',
  large: '2.5rem'
}

export const colors = {
  red: '#bf3542',
  yellow: '#ff974f',
  green: '#8c9b7a',
  blue: '#046380',
  lightBlue: '#057091',
  lightGray: '#e6e6e6'
}

export const timings = {
  easeOutQuad: 'cubic-bezier(0.25, 0.46, 0.45, 0.94)'
}

export const CONSTS = {
  headerHeight: '4.25rem'
}
