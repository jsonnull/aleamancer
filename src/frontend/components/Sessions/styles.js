// @flow
import styled from 'styled-components'
import { fontSize, fonts } from 'frontend/styles/common'

export const Container = styled.div`
  display: flex;
  flex-direction: column;
  align-items: stretch;
  flex: 1;
`

export const Body = styled.div`
  display: flex;
  flex-direction: column;
  flex: 1;
  background: ${props => props.theme.background};
  padding: 0 2rem;
`

export const Heading = styled.h1`
  font-size: ${fontSize.large};
  line-height: 1;
  font-family: ${fonts.heading};
  padding: 1rem 0;
  color: ${props => props.theme.color};
  margin: 0;
`
