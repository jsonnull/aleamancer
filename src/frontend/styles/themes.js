// @flow
import type { ThemeName } from 'common/types'
import { colors } from './common'

export type Theme = {
  name: ThemeName,
  background: string,
  backgroundSecondary: string,
  backgroundInverted: string,
  backgroundInvertedSecondary: string,
  color: string,
  colorSecondary: string,
  colorMedium: string,
  colorInverted: string,
  map: string,
  borderColor: string,
  modalOverlay: string,
  modalBackground: string
}

export const light: Theme = {
  name: 'light',
  background: '#f7f7f7',
  backgroundSecondary: '#ebe3d6',
  backgroundInverted: '#3c3c3c',
  backgroundInvertedSecondary: '#555',
  color: 'black',
  colorSecondary: '#3c3c3c',
  colorMedium: '#999',
  colorInverted: '#f7f7f7',
  map: '#bfbca5',
  borderColor: '#cdc5ba',
  modalBackground: '#f7f7f7',
  modalOverlay: 'rgba(0, 0, 0, 0.5)'
}

export const dark: Theme = {
  name: 'dark',
  background: '#3c3c3c',
  backgroundSecondary: '#2b2b2b',
  backgroundInverted: colors.blue,
  backgroundInvertedSecondary: colors.lightBlue,
  color: '#eee',
  colorSecondary: '#cdc5ba',
  colorMedium: '#999',
  colorInverted: '#eee',
  map: '#1a1a1a',
  borderColor: '#1a1a1a',
  modalBackground: '#2b2b2b',
  modalOverlay: 'rgba(0, 0, 0, 0.7)'
}
