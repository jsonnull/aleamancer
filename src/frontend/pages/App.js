// @flow
import React from 'react'
import { Route, Switch, Redirect } from 'react-router'
import { connect } from 'react-redux'
import styled from 'styled-components'
import type { State } from 'frontend/store'
import Sessions from 'frontend/containers/Sessions'
import Chat from 'frontend/containers/Chat'
import Map from 'frontend/containers/Map'
import Sidebar from 'frontend/containers/Sidebar'
import Loading from 'frontend/components/Loading'

const GameInner = styled.div`
  display: flex;
  flex-direction: row;
  align-items: stretch;
  flex: 1;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
`

const Game = () => (
  <GameInner>
    <Sidebar />
    <Map />
    <Chat />
  </GameInner>
)

type Props = {
  appIsLoading: boolean,
  userIsLoggedIn: boolean,
  location: Object
}
export class App extends React.Component<Props> {
  render() {
    const { appIsLoading, userIsLoggedIn } = this.props

    if (!userIsLoggedIn) {
      return <Redirect to="/login" />
    }

    if (appIsLoading) {
      return <Loading />
    }

    return (
      <Switch>
        <Route exact path="/sessions" component={Sessions} />
        <Route path="/g/:name/:id" component={Game} />
      </Switch>
    )
  }
}

const mapStateToProps = (state: State): Props => ({
  appIsLoading: state.ui.appIsLoading,
  userIsLoggedIn: state.ui.userIsLoggedIn,
  location: state.router.location
})

export default connect(mapStateToProps)(App)
