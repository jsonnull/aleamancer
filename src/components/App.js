// @flow
import React from 'react'
import { Switch, Route } from 'react-router'
import styled, { ThemeProvider } from 'styled-components'
import Loading from './Loading'
import Login from '../containers/Login'
import Header from '../containers/Header'
import Sessions from '../containers/Sessions'
import Settings from '../containers/Settings'
import Chat from '../containers/Chat'
import Map from '../containers/Map'
import Sidebar from '../containers/Sidebar'
import * as themes from 'styles/themes'
import { CONSTS } from 'styles/common'

const Container = styled.div`
  display: flex;
  flex-direction: column;
  align-items: stretch;
  background-color: ${props => props.theme.map};
`

const GameInner = styled.div`
  display: flex;
  flex-direction: row;
  align-items: stretch;
  flex: 1;
  position: absolute;
  top: ${CONSTS.headerHeight};
  right: 0;
  bottom: 0;
  left: 0;
`

const Game = () => (
  <GameInner>
    <Sidebar />
    <Map />
    <Chat />
  </GameInner>
)

export type Props = {
  appIsLoading: boolean,
  userIsLoggedIn: boolean,
  showSettings: boolean,
  location: Object,
  theme: Object
}
class App extends React.Component<Props> {
  render() {
    const { appIsLoading, userIsLoggedIn, showSettings, theme } = this.props

    if (appIsLoading) {
      return (
        <ThemeProvider theme={themes['light']}>
          <Loading />
        </ThemeProvider>
      )
    }

    if (!userIsLoggedIn) {
      return (
        <ThemeProvider theme={themes['light']}>
          <Login />
        </ThemeProvider>
      )
    }

    const settings = showSettings ? <Settings /> : null

    return (
      <ThemeProvider theme={theme}>
        <Container>
          <Header />
          <Switch>
            <Route exact path="/" component={Sessions} />
            <Route path="/g/:id/" component={Game} />
          </Switch>
          {settings}
        </Container>
      </ThemeProvider>
    )
  }
}

export default App
