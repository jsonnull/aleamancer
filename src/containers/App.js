// @flow
import { connect } from 'react-redux'
import App from '../components/App'
import * as themes from 'styles/themes'
import type { State } from 'store'
import type { Props } from '../components/App'

const mapStateToProps = (state: State): Props => ({
  appIsLoading: state.ui.appIsLoading,
  userIsLoggedIn: state.ui.userIsLoggedIn,
  theme: themes[state.user.preferences.theme],
  showSettings: state.ui.showSettings,
  location: state.router.location
})

export default connect(mapStateToProps)(App)
