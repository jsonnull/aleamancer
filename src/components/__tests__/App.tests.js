// @flow
import React from 'react'
import ShallowRenderer from 'react-test-renderer/shallow'
import App from '../App'
import { light } from '../../styles/themes'

describe('App component', () => {
  const renderer = new ShallowRenderer()

  it('renders correctly when loading', () => {
    renderer.render(
      <App
        appIsLoading
        userIsLoggedIn={false}
        showSettings={false}
        location={{}}
        theme={light}
      />
    )
    const tree = renderer.getRenderOutput()
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly when user is logged in', () => {
    renderer.render(
      <App
        appIsLoading={false}
        userIsLoggedIn
        showSettings={false}
        location={{}}
        theme={light}
      />
    )
    const tree = renderer.getRenderOutput()
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly when settings should be showing', () => {
    renderer.render(
      <App
        appIsLoading={false}
        userIsLoggedIn
        showSettings
        location={{}}
        theme={light}
      />
    )
    const tree = renderer.getRenderOutput()
    expect(tree).toMatchSnapshot()
  })
})
