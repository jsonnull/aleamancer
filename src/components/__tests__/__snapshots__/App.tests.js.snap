// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders correctly when loading 1`] = `
<ThemeProvider
  theme={
    Object {
      "background": "#f7f7f7",
      "backgroundInverted": "#3c3c3c",
      "backgroundInvertedSecondary": "#555",
      "backgroundSecondary": "#ebe3d6",
      "borderColor": "#cdc5ba",
      "color": "black",
      "colorInverted": "#f7f7f7",
      "colorMedium": "#999",
      "colorSecondary": "#3c3c3c",
      "map": "#bfbca5",
      "name": "light",
    }
  }
>
  <LoadingModal />
</ThemeProvider>
`;

exports[`App component renders correctly when settings should be showing 1`] = `
<ThemeProvider
  theme={
    Object {
      "background": "#f7f7f7",
      "backgroundInverted": "#3c3c3c",
      "backgroundInvertedSecondary": "#555",
      "backgroundSecondary": "#ebe3d6",
      "borderColor": "#cdc5ba",
      "color": "black",
      "colorInverted": "#f7f7f7",
      "colorMedium": "#999",
      "colorSecondary": "#3c3c3c",
      "map": "#bfbca5",
      "name": "light",
    }
  }
>
  <styled.div>
    <Connect(Header) />
    <Switch>
      <Route
        component={[Function]}
        exact={true}
        path="/"
      />
      <Route
        component={[Function]}
        path="/g/:id/"
      />
    </Switch>
    <Connect(Settings) />
  </styled.div>
</ThemeProvider>
`;

exports[`App component renders correctly when user is logged in 1`] = `
<ThemeProvider
  theme={
    Object {
      "background": "#f7f7f7",
      "backgroundInverted": "#3c3c3c",
      "backgroundInvertedSecondary": "#555",
      "backgroundSecondary": "#ebe3d6",
      "borderColor": "#cdc5ba",
      "color": "black",
      "colorInverted": "#f7f7f7",
      "colorMedium": "#999",
      "colorSecondary": "#3c3c3c",
      "map": "#bfbca5",
      "name": "light",
    }
  }
>
  <styled.div>
    <Connect(Header) />
    <Switch>
      <Route
        component={[Function]}
        exact={true}
        path="/"
      />
      <Route
        component={[Function]}
        path="/g/:id/"
      />
    </Switch>
  </styled.div>
</ThemeProvider>
`;
